src/DataStructures.jl

```diff
@@ -14,11 +14,11 @@ module DataStructures
                  ReverseOrdering, Reverse, Lt,
                  isless,
                  union, intersect, symdiff, setdiff, issubset,
-                 find, searchsortedfirst, searchsortedlast, endof, in
+                 find, searchsortedfirst, searchsortedlast, endof, in, produce
 
 
     export Deque, Stack, Queue
-    export deque, enqueue!, dequeue!, update!,iter
+    export deque, enqueue!, dequeue!, update!, forward_iter, backward_iter
```

Comment:
Although now that I say that, none of these is particularly clear for stacks, at least (because forward or backward/reverse depends on the implementation).